gateway/activation/errors.go

```diff
@@ -0,0 +1,46 @@
+package activation
+
+import (
+	"fmt"
+
+	"github.com/spacemeshos/poet/shared"
+)
+
+var (
+	ErrAtxNotFound = &AtxNotFoundError{}
+	ErrTransport   = &TransportError{}
+)
+
+// AtxNotFoundError means that ATX with given ID was not found.
+type AtxNotFoundError struct {
+	id shared.ATXID
+}
+
+func (e *AtxNotFoundError) Error() string {
+	return fmt.Sprintf("ATX %X not found", e.id)
+}
+
+func (e *AtxNotFoundError) Is(target error) bool {
+	_, ok := target.(*AtxNotFoundError)
+	return ok
+}
```

Comment:
Imo this should check if id is the same:

```suggestion
func (e *AtxNotFoundError) Is(target error) bool {
	other, ok := target.(*AtxNotFoundError)
	return ok && other.id == e.id
}
```

If one just wants to check if any atx wasn't found this can still be accomplished `errors.As` or `require.ErrorAs`:

```golang
var target AtxNotFoundError
require.ErrorsAs(err, &target) // an AtxNotFoundError with any ID
```